packages/gatsby-source-shopify/src/create-schema-customization.ts

```diff
@@ -49,6 +49,29 @@ export function createSchemaCustomization(
       }
     `)
   }
+  actions.createFieldExtension({
+    name: `selectQuantityByName`,
+    args: {
+      name: `String!`,
+    },
+    extend({ name }, fieldConfig) {
+      return {
+        async resolve(source, args, context, info): Promise<number> {
+          const resolver = fieldConfig.resolve || context.defaultFieldResolver
+          const quantities = await resolver(source, args, context, info)
 
+          if (quantities && Array.isArray(quantities)) {
+            for (const quantity of quantities) {
+              if (quantity?.name === name) {
```

Comment:
This looks like it could be `quantities.find(quantity?.name === name)`

Reply:
Addressed in https://github.com/gatsbyjs/gatsby/pull/39082/commits/2feb7371e217fcb915581ebf528b26b7ac43e12d